esteid/mobileid/base.py

```diff
@@ -0,0 +1,423 @@
+        "no_session_code": "Session {} does not exist",
+        "action_not_completed": "Action for session {} has not completed yet",
+        "unexpected_state": "Unexpected state {}",
+        "unexpected_end_result": "Unexpected end result {}",
+        "signature_mismatch": "Signature mismatch",
+        "timed_out": "Connection timed out, retry later",
+        "invalid_credentials": "Authentication failed: Check rp_uuid and verify the ip of the "
+        "server has been added to the service contract",
+        "unsupported_client": "The client is not supported",
+        "maintenance": "System is under maintenance, retry later",
+        "proxy_error": "Proxy error {}, retry later",
+        "http_error": "Invalid response code(status_code: {0}, body: {1})",
+        "invalid_signature_algorithm": "Invalid signature algorithm {}",
+    }
+
+    END_RESULT_MESSAGES = {
+        EndResults.OK: "Successfully authenticated with Mobile-ID",
+        EndResults.USER_CANCELLED: "User canceled the Mobile-ID request",
+        EndResults.TIMEOUT: "Mobile-ID request timed out",
+        EndResults.NOT_MID_CLIENT: "Not a Mobile-ID client",
+    }
+
+    class Actions:
+        AUTH = "/authentication"
+        SIGN = "/signature"
+        SESSION_STATUS = "{action_type}/session/{session_id}"
+
+    def __init__(self, rp_uuid, rp_name, api_root):
+        self.rp_uuid = rp_uuid  # type: UUID
+        self.rp_name = rp_name  # type: str
+        self.api_root = api_root
+
+        self.session = requests.Session()
+
+    def get_certificate(self, id_code, phone_number) -> bytes:
+        """
+
+        :param id_code:
+        :param phone_number:
+        :return: certificate
+        """
+        assert id_code
+        assert phone_number
+        endpoint = "/certificate"
+
+        result = self.invoke(
+            endpoint,
+            method="POST",
+            data=self.rp_params({"phoneNumber": phone_number, "nationalIdentityNumber": id_code}),
+        )
+
+        status = result["result"]
+        if status == "OK":
+            return base64.b64decode(result["cert"])
+        if status == "NOT_FOUND":
+            raise UserNotRegistered(
+                "User with phone number {phone} and ID code {id_code} not found".format(
+                    phone=phone_number, id_code=id_code
+                )
+            )
+        raise MobileIDError("Unknown response format")
+
+    def authenticate(self, id_code, phone_number, message=None, language=None, hash_type=HASH_SHA256):
+        """Initiate an authentication session
+
+        https://github.com/SK-EID/MID#32-initiating-signing-and-authentication
+
+        Difference between auth and signing is the hash: for auth, it's just a random string of the corresponding size
+        (the size of a hash produced by selected algorithm) - 32 for SHA256
+
+        https://github.com/SK-EID/MID#23-creating-the-hash
+
+        :param str id_code: National identity number
```

Comment:
Why doc string typing rather than parameter typing?

Reply:
It dates back to the times when Python 2 was supported. :) But yeah makes sense to move type hints to the declaration